### PR TITLE
refactor:  repo counts for SearchRepositoryByName func

### PR DESCRIPTION
**Pull request by appleboy**

### CODE REVIEW

**Comment by ethantkoenig** [comment]
@appleboy It's not clear to me why this is an improvement. Could you briefly explain why?

File: models/repo.go [hunk 1]
```diff
@@ -1888,15 +1888,21 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, _ in
...
+		if err != nil {
+			return nil, 0, fmt.Errorf("Count: %v", err)
+		}
 	} else {
 		sess = x.Where(cond)
-	}
-
-	var countSess xorm.Session
-	countSess = *sess
-	count, err := countSess.Count(new(Repository))
```
Comment (appleboy) [inline models/repo.go:1897]: @ethantkoenig remove countSess session.

**Comment by appleboy** [comment]
@ethantkoenig just like this PR https://github.com/go-gitea/gitea/pull/1038/files

**Comment by lunny** [comment]
@ethantkoenig 
`countSess = *sess` is dangous for XORM. I recommend two ways to do that:
One is use builder, another is use `func createSession()`, but not this one.

**Comment by appleboy** [comment]
@lunny Thanks for your explanation.

**Comment by lunny** [comment]
LGTM

**Comment by ethantkoenig** [comment]
LGTM

**Pull request merged**